lib/git-util.js

```diff
@@ -21,3 +21,69 @@ exports.hasUpstream = async () => {
 	const {stdout} = await execa('git', ['status', '--short', '--branch', '--porcelain=2']);
 	return /^# branch\.upstream [\w\-/]+$/m.test(stdout);
 };
+
+exports.currentBranch = () => execa.stdout('git', ['symbolic-ref', '--short', 'HEAD']);
+
+exports.isWorkingTreeClean = async () => {
+	try {
+		const {stdout: status} = await execa('git', ['status', '--porcelain']);
+		if (status !== '') {
+			return false;
+		}
+
+		return true;
+	} catch (_) {
+		return false;
+	}
+};
+
+exports.isRemoteHistoryClean = async () => {
+	let history;
+	try { // Gracefully handle no remote set up.
+		history = await execa.stdout('git', ['rev-list', '--count', '--left-only', '@{u}...HEAD']);
+	} catch (_) {}
+
+	if (history && history !== '0') {
+		return false;
+	}
+
+	return true;
+};
+
+exports.verifyRemoteIsValid = async () => {
+	try {
+		await execa('git', ['ls-remote', 'origin', 'HEAD']);
+	} catch (error) {
+		throw error.stderr.replace('fatal:', 'Git fatal error:');
+	}
+};
+
+exports.fetchRemote = () => execa('git', ['fetch']);
+
+exports.tagExists = async tagName => {
```

Comment:
Maybe we should be explicit whether this checks if the tag exists locally, remotely, or both?